runtime/lib/capi/ExecutionContext.hpp

```diff
@@ -262,10 +263,19 @@ class ExecutionContext final {
     std::unique_ptr<MemoryManager> memory_man_ptr{nullptr};
     std::unique_ptr<PythonInterpreterGuard> py_guard{nullptr};
 
+    // PRNG
+    uint32_t *seed = nullptr;
+    std::mt19937 gen;
+
   public:
-    explicit ExecutionContext() : initial_tape_recorder_status(false)
+    explicit ExecutionContext(uint32_t *seed = nullptr) : initial_tape_recorder_status(false)
```

Comment:
This is what I mean (untested)
```suggestion
    explicit ExecutionContext(uint32_t *seed = nullptr) : initial_tape_recorder_status(false), seed(seed)
```

Reply:
Oh I see, let's try

Reply:
done